release/tar/linux/opensearch-tar-build.sh

```diff
@@ -0,0 +1,159 @@
+# permissions and limitations under the License.
+#
+# About:        This shell script generate an opensearch tarball with all the provided plugins installed and ready to use. 
+#               A manifest file and right permissions set up is essential to download the required artifacts
+# Dependencies: yq (More info: https://github.com/mikefarah/yq/tree/v4.4.1#install)
+# Usage: .      /opensearch-tar-build.sh -f <path/to/manifest/file/>
+#
+
+set -e
+
+function usage() {
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-f MANIFEST_FILE\tPath/to/manifest.yml file"
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-p PLATFORM\tSpecify the platform. eg: linux, macos, windows. Currently only supports and defaults to linux"
+    echo -e "-a ARCHITECTURE\tSpecify the architecture, e.g. x64, arm64. Currently only supports x64 and arm64. Defaults to x64"
+    echo -e "-h help"
+}
+
+while getopts ":ha:p:f:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        p)
+            PLATFORM=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        a)
+            ARCHITECTURE=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+            ;;
+        f)
+            MANIFEST_FILE=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$PLATFORM" ]; then
+    PLATFORM="linux"
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: This platform is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$ARCHITECTURE" ]; then
+    ARCHITECTURE="x64"
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] || [ "$ARCHITECTURE" != "arm64" ]; then
+    echo "error: This architecture is not supported yet!"
+    usage
+    exit 1
+fi
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "Please specify the right path to manifest file"
+    usage
+    exit 1
+fi
+REPO_ROOT=`git rev-parse --show-toplevel`
+VERSION=`yq eval '.version' $MANIFEST_FILE`
+WORKING_DIR="opensearch-working-directory"
+OPENSEARCH_CORE_URL=`yq eval '.products.opensearch.opensearch-min' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+OPENSEARCH_PLUGINS_URLS=`yq eval '.products.opensearch.plugins' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+LIBS=`yq eval '.products.opensearch.libs' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+
+# Download plugins to local 
+mkdir -p ${WORKING_DIR}/plugins_downloads
+IFS=$'\n'
+cd ${WORKING_DIR}/plugins_downloads
+for plugin in ${OPENSEARCH_PLUGINS_URLS}; do
+  aws s3 cp $plugin .
+done
+
+# Download k-NN lib
+aws s3 cp $LIBS .
+unzip opensearch-knnlib-*-$PLATFORM-$ARCHITECTURE.zip
+
+cd ..
+
+# Download OpenSearch core TAR
+wget -q ${OPENSEARCH_CORE_URL}
+tar -xzf opensearch-min-*
+rm *tar.gz
```

Comment:
This tightly couples the script to the filename in the manifest. I'd recommend this instead:

```
wget -q ${OPENSEARCH_CORE_URL} -O opensearch.tgz
tar -xzf opensearch.tgz
rm opensearch.tgz
```

That way there is no way for a glob to accidentally pick up an unrelated tarball, and it doesn't matter if the manifest specifies a differently-named file (for example, I might want to build a bundle from `camerski-testing-opensearch.tar.gz`)

Reply:
Makes sense. I did that in CDK not sure why I skipped here. Will make the changes.

Reply:
Added